combined.js

```diff
@@ -0,0 +1,159 @@
+// ICEBOX - gives user the option to play either game
+
+const { reverse } = require("dns");
+const readline = require("readline");
+const { start } = require("repl");
+const rl = readline.createInterface(process.stdin, process.stdout);
+
+function ask(questionText) {
+  return new Promise((resolve, reject) => {
+    rl.question(questionText, resolve);
+  });
+}
+
+combined()
+
+async function combined() {
+    console.log("Hello! We have two game options: Either A) You think of a number and the computer tries to guess it, or B) The computer thinks of a number and you try to guess it. Which would you like to play?")
+    let game = await ask("Please input A or B => ")
+    if (game === "A") {
+        start()
+    } else {
+        reverse()
+    }
+
+
+async function start() {
+    console.log(
+      "Let's play a game where you (human) make up a number, and I (computer) try to guess it. The lowest possible number is 1."
+    );
+  
+    let lowest = 1;
+  
+    // Story 6: "Extend the Guess Range"
+    let highest = await ask(
+      "Please enter a number that we'll use as the highest possible guess => "
+    );
+    highest = parseInt(highest);
+  
+    let secretNumber = await ask(
+      "Ok, now enter a secret number between 1 and " +
+        highest +
+        ". (I promise I won't look.)\n"
+    );
+    console.log("YOUR SECRET NUMBER: " + secretNumber);
+  
+    // Story 1: "Pick a Number, Any Number"
+    function randomNum() {
+      return Math.floor(Math.random() * highest + lowest);
+    }
+  
+    let guess = randomNum();
+  
+    let response = await ask("Is your number " + guess + "? yes or no? => ");
+  
+    // Story 2: Computer guessed correctly
+    if (response === "yes") {
+      console.log("Wow there was a 1 in 100 chance of that. Let's go eat cake!");
+      process.exit();
+    }
+  
+    // Story 3: Computer guessed wrong
+    else {
+      response = await ask(
+        "I'll try again! Is your number higher (key in 'H') or lower (key in 'L')? => "
+      );
+      let calledTimes = 1; // ICEBOX - counting number of guesses
+  
+      // Stories 4 and 5: "Modify Your Guess Range" and "Make it Smarter"
+  
+      function guessHigher() {
+        guess = Math.round((highest + guess) / 2); // using *round* here (as opposed to *floor*) so it'll *include* the high range as a possibility
+        return guess;
+      }
+  
+      function guessLower() {
+        guess = Math.floor((guess + lowest) / 2);
+        return guess;
+      }
+  
+      while (response !== "yes" && response !== "Yes" && response !== "YES") {
```

Comment:
A hint for your cheat detector since I know you were having difficulty with it: in the conditional logic for your if statement ```if (response === "H" || response === "h")```, you might consider 1) switching to AND instead of OR 2) adding another check to see if the current guess is higher or lower than the secret number.